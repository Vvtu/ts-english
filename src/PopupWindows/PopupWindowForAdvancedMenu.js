import React, { PureComponent } from 'react';

import CancelIcon from './svg/CancelIcon';
import ClearAllInfo from './svg/ClearAllInfo';

import './PopupWindows.css';

import { ANIMATION_INTERVAL, ICON_SIZE } from '../Constants/constants';

const TRANSITION = 'opacity ' + ANIMATION_INTERVAL + 's ease';

const noop = () => {};

class PopupWindowForAdvancedMenu extends PureComponent {
	constructor(props) {
		super(props);
		this.state = {};
	}

	componentDidMount() {
		requestAnimationFrame(() => {
			this.setState({ increaseOpacity: true });
		});
	}

	menuRow = (menuItem) => (
		<div
			className="popup__row"
			onClick={(e) => this.handleClicked(e, menuItem.handleClickedFunc)}
			onDoubleClick={(e) => this.handleClicked(e, menuItem.handleClickedFunc)}
		>
			<menuItem.Icon {...menuItem.props} />
			<div className="popup__gap" />
			<div>{menuItem.label}</div>
		</div>
	);

	handleClicked = (e, func) => {
		e.preventDefault();
		e.stopPropagation();
		this.setState({ increaseOpacity: false });
		setTimeout(() => {
			func();
		}, ANIMATION_INTERVAL * 1000);
		return false;
	};

	render() {
		const styles = getComputedStyle(document.documentElement);
		const greenColor = String(styles.getPropertyValue('--english-text-color')).trim();
		const redColor = String(styles.getPropertyValue('--danger-color')).trim();
		const whiteColor = String(styles.getPropertyValue('--base-text-color')).trim();
		const blackColor = String(styles.getPropertyValue('--background-color')).trim();

		const {
			handleHideItemClicked,
			handleUnhideAllItemsClicked,
			handleDictClicked,
			handleClosePopupClicked,
			handleShowSetVoicePopupClicked,
		} = this.props;

		// console.log('this.state.increaseOpacity = ', this.state.increaseOpacity);

		return (
			<div
				style={{
					opacity: this.state.increaseOpacity === true ? '1' : '0',
					transition: TRANSITION,
				}}
				onClick={(e) => this.handleClicked(e, handleClosePopupClicked)}
				onDoubleClick={(e) => this.handleClicked(e, handleClosePopupClicked)}
			>
				<div className="popup__full_screen_div_opacity" />
				<div className="popup__full_screen_div">
					<div className="popup__window">
						<div className="popup__cancel">
							<CancelIcon
								fill={whiteColor}
								height={16}
								onClick={(e) => this.handleClicked(e, handleClosePopupClicked)}
								onDoubleClick={(e) => this.handleClicked(e, handleClosePopupClicked)}
								width={16}
							/>
						</div>
						<div className="popup__list">
							{this.menuRow({
								label: 'Advanced:',
								Icon: CancelIcon,
								props: { height: ICON_SIZE, width: ICON_SIZE, fill: blackColor },
								handleClickedFunc: noop,
							})}

							{this.menuRow({
								label: 'hide this item',
								Icon: CancelIcon,
								props: { height: ICON_SIZE, width: ICON_SIZE, fill: redColor },
								handleClickedFunc: handleHideItemClicked,
							})}

							{this.menuRow({
								label: 'reset all info',
								Icon: ClearAllInfo,
								props: { height: ICON_SIZE, width: ICON_SIZE, fill: greenColor },
								handleClickedFunc: handleUnhideAllItemsClicked,
							})}

							{this.menuRow({
								label: 'dict #1',
								Icon: () => <div style={{ width: ICON_SIZE, color: greenColor }}>D1</div>,
								props: {},
								handleClickedFunc: () => handleDictClicked(1),
							})}

							{this.menuRow({
								label: 'dict #2',
								Icon: () => <div style={{ width: ICON_SIZE, color: greenColor }}>D2</div>,
								props: {},
								handleClickedFunc: () => handleDictClicked(2),
							})}
							{this.menuRow({
								label: 'set voice',
								Icon: () => <div style={{ width: ICON_SIZE, color: greenColor }}>D2</div>,
								props: {},
								handleClickedFunc: () => handleShowSetVoicePopupClicked(),
							})}
						</div>
					</div>
				</div>
			</div>
		);
	}
}

export default PopupWindowForAdvancedMenu;
