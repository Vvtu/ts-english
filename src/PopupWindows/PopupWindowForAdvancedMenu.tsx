import React, { PureComponent, SyntheticEvent } from "react";

import CancelIcon from "./svg/CancelIcon";
import ClearAllInfo from "./svg/ClearAllInfo";

import {
  ANIMATION_INTERVAL,
  ICON_SIZE
} from "../Constants/constants";

const TRANSITION = "opacity " + ANIMATION_INTERVAL + "s ease";

const noop = () => {};

export interface Props {
  handleHideItemClicked(): void;
  handleUnhideAllItemsClicked(): void;
  handleDictClicked(number: number): void;
  handleClosePopupClicked(): void;
  handleShowSetVoicePopupClicked(): void;

  // selectedVoiceIndex: string | null;
  // voicesArray: Voice[];
  // handleClosePopupClicked(): void;
  // handleVoiceDidSelect(index: string): void;
}

interface State {
  increaseOpacity: boolean;
}

class PopupWindowForAdvancedMenu extends PureComponent<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { increaseOpacity: false };
  }

  componentDidMount() {
    requestAnimationFrame(() => {
      this.setState({ increaseOpacity: true });
    });
  }

  menuRow = ({
    Icon,
    label,
    menuProps,
    handleClickedFunc
  }: {
    Icon: any;
    label: string;
    menuProps:
      | {
          height: string;
          width: string;
          fill: string;
        }
      | {};
    handleClickedFunc(): void;
  }) => (
    <div
      className="popup__row"
      onClick={(e: SyntheticEvent) =>
        this.handleClicked(e, handleClickedFunc)
      }
      onDoubleClick={(e: SyntheticEvent) =>
        this.handleClicked(e, handleClickedFunc)
      }
    >
      <Icon {...menuProps} />
      <div className="popup__gap" />
      <div>{label}</div>
    </div>
  );

  handleClicked = (e: SyntheticEvent, func: () => void) => {
    e.stopPropagation();
    this.setState({ increaseOpacity: false });
    setTimeout(() => {
      func();
    }, ANIMATION_INTERVAL * 1000);
    return false;
  };

  render() {
    const styles = getComputedStyle(document.documentElement);
    const greenColor = String(
      styles.getPropertyValue("--english-text-color")
    ).trim();
    const redColor = String(
      styles.getPropertyValue("--danger-color")
    ).trim();
    const whiteColor = String(
      styles.getPropertyValue("--base-text-color")
    ).trim();
    const blackColor = String(
      styles.getPropertyValue("--background-color")
    ).trim();

    const {
      handleHideItemClicked,
      handleUnhideAllItemsClicked,
      handleDictClicked,
      handleClosePopupClicked,
      handleShowSetVoicePopupClicked
    } = this.props;

    // console.log('this.state.increaseOpacity = ', this.state.increaseOpacity);

    return (
      <div
        style={{
          opacity: this.state.increaseOpacity === true ? 1 : 0,
          transition: TRANSITION
        }}
        onClick={(e: SyntheticEvent) =>
          this.handleClicked(e, handleClosePopupClicked)
        }
        onDoubleClick={(e: SyntheticEvent) =>
          this.handleClicked(e, handleClosePopupClicked)
        }
      >
        <div className="popup__full_screen_div_opacity" />
        <div className="popup__full_screen_div">
          <div className="popup__window">
            <div className="popup__cancel">
              <CancelIcon
                fill={whiteColor}
                height={16}
                onClick={(e: SyntheticEvent) =>
                  this.handleClicked(e, handleClosePopupClicked)
                }
                onDoubleClick={(e: SyntheticEvent) =>
                  this.handleClicked(e, handleClosePopupClicked)
                }
                width={16}
              />
            </div>
            <div className="popup__list">
              {this.menuRow({
                label: "Advanced:",
                Icon: CancelIcon,
                menuProps: {
                  height: ICON_SIZE,
                  width: ICON_SIZE,
                  fill: blackColor
                },
                handleClickedFunc: noop
              })}

              {this.menuRow({
                label: "hide this item",
                Icon: CancelIcon,
                menuProps: {
                  height: ICON_SIZE,
                  width: ICON_SIZE,
                  fill: redColor
                },
                handleClickedFunc: handleHideItemClicked
              })}

              {this.menuRow({
                label: "reset all info",
                Icon: ClearAllInfo,
                menuProps: {
                  height: ICON_SIZE,
                  width: ICON_SIZE,
                  fill: greenColor
                },
                handleClickedFunc: handleUnhideAllItemsClicked
              })}

              {this.menuRow({
                label: "dict #1",
                Icon: () => (
                  <div
                    style={{ width: ICON_SIZE, color: greenColor }}
                  >
                    D1
                  </div>
                ),
                menuProps: {},
                handleClickedFunc: () => handleDictClicked(1)
              })}

              {this.menuRow({
                label: "dict #2",
                Icon: () => (
                  <div
                    style={{ width: ICON_SIZE, color: greenColor }}
                  >
                    D2
                  </div>
                ),
                menuProps: {},
                handleClickedFunc: () => handleDictClicked(2)
              })}
              {this.menuRow({
                label: "set voice",
                Icon: () => (
                  <div
                    style={{ width: ICON_SIZE, color: greenColor }}
                  >
                    D2
                  </div>
                ),
                menuProps: {},
                handleClickedFunc: () =>
                  handleShowSetVoicePopupClicked()
              })}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default PopupWindowForAdvancedMenu;
