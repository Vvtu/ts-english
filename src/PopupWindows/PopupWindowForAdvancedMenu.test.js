import React from 'react';
import renderer from 'react-test-renderer';

import PopupWindowForAdvancedMenu from './PopupWindowForAdvancedMenu';

describe('PopupWindowForAdvancedMenu test   ', () => {
    it('test 1', () => {
        const component = renderer.create(<PopupWindowForAdvancedMenu  />);
        const result = component.toJSON();
        expect(result).toMatchSnapshot();
    });
});
