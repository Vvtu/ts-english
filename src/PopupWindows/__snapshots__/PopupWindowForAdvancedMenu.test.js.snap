// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupWindowForAdvancedMenu test    test 1 1`] = `
<div
  onClick={[Function]}
  onDoubleClick={[Function]}
  style={
    Object {
      "opacity": "0",
      "transition": "opacity 0.5s ease",
    }
  }
>
  <div
    className="popup__full_screen_div_opacity"
  />
  <div
    className="popup__full_screen_div"
  >
    <div
      className="popup__window"
    >
      <div
        className="popup__cancel"
        onClick={undefined}
      >
        <svg
          fill=""
          height={16}
          onClick={[Function]}
          onDoubleClick={[Function]}
          viewBox="0 0 32 32"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="32"
            width="32"
          />
          <polygon
            fill=""
            points="2,26 6,30 16,20 26,30 30,26 20,16 30,6 26,2 16,12 6,2 2,6 12,16"
          />
        </svg>
      </div>
      <div
        className="popup__list"
      >
        <div
          className="popup__row"
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <svg
            fill=""
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="32"
              width="32"
            />
            <polygon
              fill=""
              points="2,26 6,30 16,20 26,30 30,26 20,16 30,6 26,2 16,12 6,2 2,6 12,16"
            />
          </svg>
          <div
            className="popup__gap"
          />
          <div>
            Advanced:
          </div>
        </div>
        <div
          className="popup__row"
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <svg
            fill=""
            height="32"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="32"
              width="32"
            />
            <polygon
              fill=""
              points="2,26 6,30 16,20 26,30 30,26 20,16 30,6 26,2 16,12 6,2 2,6 12,16"
            />
          </svg>
          <div
            className="popup__gap"
          />
          <div>
            hide this item
          </div>
        </div>
        <div
          className="popup__row"
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <svg
            fill=""
            height="32"
            viewBox="0 0 16 16"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fillRule="evenodd"
              id="Icons with numbers"
            >
              <g
                fill=""
                id="Group"
                transform="translate(-432 -432)"
              >
                <path
                  d="M438.972188,442 L438.972188,433 L447.922128,433 L444.938815,436 C446.430471,437 448.419347,439.5 447.922128,442 C447.424909,445 445.436034,447 440.961064,447 C435.491657,447 434,442.5 434,442 C436.983313,444.5 438.972188,445 440.961064,445 C442.949939,445 443.944377,443.5 443.944377,442 C443.944377,441 443.447158,440 441.955502,439 L438.972188,442 L438.972188,442 Z M438.972188,442"
                  id="Shape copy"
                  transform="matrix(-1 0 0 1 882 0)"
                />
              </g>
            </g>
          </svg>
          <div
            className="popup__gap"
          />
          <div>
            reset all info
          </div>
        </div>
        <div
          className="popup__row"
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <div
            style={
              Object {
                "color": "",
                "width": "32",
              }
            }
          >
            D1
          </div>
          <div
            className="popup__gap"
          />
          <div>
            dict #1
          </div>
        </div>
        <div
          className="popup__row"
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <div
            style={
              Object {
                "color": "",
                "width": "32",
              }
            }
          >
            D2
          </div>
          <div
            className="popup__gap"
          />
          <div>
            dict #2
          </div>
        </div>
        <div
          className="popup__row"
          onClick={[Function]}
          onDoubleClick={[Function]}
        >
          <div
            style={
              Object {
                "color": "",
                "width": "32",
              }
            }
          >
            D2
          </div>
          <div
            className="popup__gap"
          />
          <div>
            set voice
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
